# HG changeset patch
# User Rik <rik@octave.org>
# Date 1442940073 25200
#      Tue Sep 22 09:41:13 2015 -0700
# Node ID b6ae0ef9327e76d309858ccd344fb3c17cf16172
# Parent  afdb856e44f19c16322d6a81d6ef610c1b8fc787
Display bottom Qt statusbar on figure based on "toolbar" property (bug #46025).

* Figure.cc (Figure): Adjust bottom of window height (boffset) if m_statusBar
is visible.  Use m_statusBar->hide() to hide bottom status bar when "toolbar"
property is "none".

* Figure.cc (showFigureToolbar): Also show or hide m_statusBar widget based on
"toolbar" property.

diff --git a/libgui/graphics/Figure.cc b/libgui/graphics/Figure.cc
--- a/libgui/graphics/Figure.cc
+++ b/libgui/graphics/Figure.cc
@@ -134,30 +134,35 @@ Figure::Figure (const graphics_object& g
 
   // Enable mouse tracking
   m_container->setMouseTracking (true);
   foreach (QWidget* w, m_container->findChildren<QWidget*> ())
     { w->setMouseTracking (true); }
   
   // Status bar
   m_statusBar = win->statusBar ();
-  m_statusBar->setVisible (true);
-  int boffset = m_statusBar->sizeHint ().height ();
-  
+  int boffset = 0;
 
   // Toolbar and menubar
   createFigureToolBarAndMenuBar ();
   int toffset = 0;
 
   if (fp.toolbar_is ("figure") ||
       (fp.toolbar_is ("auto") && fp.menubar_is ("figure") &&
        ! hasUiControlChildren (fp)))
-    toffset += m_figureToolBar->sizeHint ().height ();
+    {
+      toffset += m_figureToolBar->sizeHint ().height ();
+      boffset += m_statusBar->sizeHint ().height ();
+    } 
   else
-    m_figureToolBar->hide ();
+    {
+      m_figureToolBar->hide ();
+      m_statusBar->hide ();
+    }
+
   if (fp.menubar_is ("figure") || hasUiMenuChildren (fp))
     toffset += m_menuBar->sizeHint ().height () + 1;
   else
     m_menuBar->hide ();
 
   m_innerRect = boundingBoxToRect (fp.get_boundingbox (true));
   m_outerRect = boundingBoxToRect (fp.get_boundingbox (false));
 
@@ -497,27 +502,29 @@ Figure::update (int pId)
   m_blockUpdates = false;
 }
 
 void
 Figure::showFigureToolBar (bool visible)
 {
   if ((! m_figureToolBar->isHidden ()) != visible)
     {
-      int dy = m_figureToolBar->sizeHint ().height ();
+      int dy1 = m_figureToolBar->sizeHint ().height ();
+      int dy2 = m_statusBar->sizeHint ().height ();
       QRect r = qWidget<QWidget> ()->geometry ();
 
       if (! visible)
-        r.adjust (0, dy, 0, 0);
+        r.adjust (0, dy1, 0, -dy2);
       else
-        r.adjust (0, -dy, 0, 0);
+        r.adjust (0, -dy1, 0, dy2);
 
       m_blockUpdates = true;
       qWidget<QWidget> ()->setGeometry (r);
       m_figureToolBar->setVisible (visible);
+      m_statusBar->setVisible (visible);
       m_blockUpdates = false;
 
       updateBoundingBox (false);
     }
 }
 
 void
 Figure::showMenuBar (bool visible)
